File: LHDS.Core/Models/Foundations/DatasetObjects/DatasetObject.cs
// ---------------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------------

using System;
using System.Collections.Generic;
using LHDS.Core.Models.Bases;
using LHDS.Core.Models.Foundations.ColumnDefinitions;
using LHDS.Core.Models.Foundations.DataSetSpecifications;

namespace LHDS.Core.Models.Foundations.DataSetObjects
{
    public class DataSetObject : IKey, IAudit
    {
        public Guid Id { get; set; }
        public Guid DataSetSpecificationId { get; set; }
        public string SupplierObjectName { get; set; }
        public string OurObjectName { get; set; }
        public string ObjectDescription { get; set; }
        public string InterchangeProtocol { get; set; }
        public string PushOrPull { get; set; }
        public string DeletionHandling { get; set; }
        public bool IsSubmissionHeaderObject { get; set; }
        public bool IsTransactionLog { get; set; }
        public string CreatedBy { get; set; }
        public string UpdatedBy { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }
        public DateTimeOffset CreatedDate { get; set; }

        public DataSetSpecification DataSetSpecification { get; set; }
        public List<ObjectColumn> DataSetObjects { get; set; } = new List<ObjectColumn>();

    }
}

File: LHDS.AdminPortal.Web/ClientApp/src/models/dataSetObjects/dataSetObject.ts
import { Guid } from 'guid-typescript';

export class DataSetObject {
    public id: Guid;
    public dataSetSpecificationId: Guid;
    public SupplierObjectName: string;
    public OurObjectName: string;
    public ObjectDescription: string;
    public InterchangeProtocol: boolean;
    public PushOrPull: string;
    public DeletionHandling: string;
    public IsSubmissionHeaderObject: boolean;
    public IsTransactionLog: boolean
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;

    constructor(dataSetObjects: any) {
        this.id = dataSetObjects.id ? Guid.parse(dataSetObjects.id) : Guid.parse(Guid.EMPTY);
        this.dataSetSpecificationId = dataSetObjects.dataSetSpecificationId;
        this.SupplierObjectName = dataSetObjects.SupplierObjectName;
        this.OurObjectName = dataSetObjects.ourSpecificationVersion;
        this.ObjectDescription = dataSetObjects.ObjectDescription;
        this.InterchangeProtocol = dataSetObjects.InterchangeProtocol;
        this.PushOrPull = dataSetObjects.PushOrPull;
        this.DeletionHandling = dataSetObjects.DeletionHandling;
        this.IsSubmissionHeaderObject = dataSetObjects.IsSubmissionHeaderObject;
        this.IsTransactionLog = dataSetObjects.IsTransactionLog;
        this.createdDate = dataSetObjects.createdDate;
        this.createdBy = dataSetObjects.createdBy;
        this.createdDate = new Date(dataSetObjects.createdDate);
        this.updatedBy = dataSetObjects.updatedBy;
        this.updatedDate = new Date(dataSetObjects.updatedDate);
    }
}